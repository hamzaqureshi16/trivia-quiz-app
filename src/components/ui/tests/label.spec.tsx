import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Label } from '../label';

describe('Label Component', () => {
  it('renders the Label with children', () => {
    render(<Label>Test Label</Label>);
    expect(screen.getByText('Test Label')).toBeInTheDocument();
  });

  it('applies the htmlFor prop correctly', () => {
    render(<Label htmlFor="test-input">Test Label</Label>);
    const labelElement = screen.getByText('Test Label');
    expect(labelElement).toHaveAttribute('for', 'test-input');
  });

  it('applies custom class names', () => {
    render(<Label className="custom-class">Styled Label</Label>);
    const labelElement = screen.getByText('Styled Label');
    expect(labelElement).toHaveClass('custom-class');
  });

  it('matches the snapshot', () => {
    const { container } = render(<Label htmlFor="snapshot-input">Snapshot Label</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
