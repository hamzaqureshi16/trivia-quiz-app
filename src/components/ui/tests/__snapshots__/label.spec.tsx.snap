// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component matches the snapshot 1`] = `
<label
  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
  data-slot="label"
  for="snapshot-input"
>
  Snapshot Label
</label>
`;
